pulsar-broker/src/main/java/org/apache/pulsar/common/naming/FlowOrQpsEquallyDivideBundleSplitAlgorithm.java

```diff
@@ -0,0 +1,131 @@
+        double diffThreshold = bundleSplitOption.getFlowOrQpsDifferenceThresholdPercentage() / 100.0;
+        long loadBalancerNamespaceBundleMaxBandwidthBytes = bundleSplitOption
+                .getLoadBalancerNamespaceBundleMaxBandwidthMbytes() * MBytes;
+
+
+        return service.getOwnedTopicListForNamespaceBundle(bundle).thenCompose(topics -> {
+            if (topics == null || topics.size() <= 1) {
+                return CompletableFuture.completedFuture(null);
+            }
+
+            double bundleThroughput = 0;
+            double bundleMsgRate = 0;
+            Map<Long, TopicInfo> topicInfoMap = new HashMap<>();
+            List<Long> topicHashList = new ArrayList<>(topics.size());
+            for (String topic : topics) {
+                TopicStatsImpl topicStats = topicStatsMap.get(topic);
+                if (topicStats == null) {
+                    continue;
+                }
+                double msgRateIn = topicStats.getMsgRateIn();
+                double msgRateOut = topicStats.getMsgRateOut();
+                double msgThroughputIn = topicStats.getMsgThroughputIn();
+                double msgThroughputOut = topicStats.getMsgThroughputOut();
+                double msgRate = msgRateIn + msgRateOut;
+                double throughput = msgThroughputIn + msgThroughputOut;
+                if (msgRate <= 0 && throughput <= 0) {
+                    // Skip empty topic
+                    continue;
+                }
+
+                Long hashCode = bundle.getNamespaceBundleFactory().getLongHashCode(topic);
+                topicHashList.add(hashCode);
+                topicInfoMap.put(hashCode, new TopicInfo(topic, msgRate, throughput));
+                bundleThroughput += throughput;
+                bundleMsgRate += msgRate;
+            }
+
+            if (topicInfoMap.size() < 2
+                    || (bundleMsgRate < (loadBalancerNamespaceBundleMaxMsgRate * (1 + diffThreshold))
+                    && bundleThroughput < (loadBalancerNamespaceBundleMaxBandwidthBytes * (1 + diffThreshold)))) {
+                return CompletableFuture.completedFuture(null);
+            }
+            Collections.sort(topicHashList);
+
+
+            List<Long> splitResults = new ArrayList<>();
+            double bundleMsgRateTmp = topicInfoMap.get(topicHashList.get(0)).msgRate;
+            double bundleThroughputTmp = topicInfoMap.get(topicHashList.get(0)).throughput;
+
+            for (int i = 1; i < topicHashList.size(); i++) {
+                long topicHashCode = topicHashList.get(i);
+                double msgRate = topicInfoMap.get(topicHashCode).msgRate;
+                double throughput = topicInfoMap.get(topicHashCode).throughput;
+
+                if (bundleMsgRateTmp > loadBalancerNamespaceBundleMaxMsgRate
+                        || bundleThroughputTmp > loadBalancerNamespaceBundleMaxBandwidthBytes
+                        || (bundleMsgRateTmp + msgRate) > loadBalancerNamespaceBundleMaxMsgRate
```

Comment:
`(bundleMsgRateTmp + msgRate) > loadBalancerNamespaceBundleMaxMsgRate` includes `bundleMsgRateTmp > loadBalancerNamespaceBundleMaxMsgRate`, so the latter one can be removed.

So is bundleThroughputTmp.

Reply:
Fixed